app/restore_to.go

```diff
@@ -0,0 +1,175 @@
+				c.String("backup-url"), c.String("output-file"), c.String("image-format"))
+			if err := restoreTo(c); err != nil {
+				logrus.Fatalf("Error running restore to file command: %v", err)
+			}
+		},
+	}
+}
+
+func imageFormatSupported(desiredFormat string) bool {
+	for _, supportedFormat := range SupportedImageFormats {
+		if desiredFormat == supportedFormat {
+			return true
+		}
+	}
+	return false
+}
+
+func restoreTo(c *cli.Context) error {
+	imageFormat := c.String("image-format")
+	if !imageFormatSupported(imageFormat) {
+		return fmt.Errorf("unsupported image format: %s", imageFormat)
+	}
+
+	backupURL := c.String("backup-url")
+	if backupURL == "" {
+		return fmt.Errorf("backup-url must be provided")
+	}
+
+	if err := backupstore.RestoreDeltaBlockBackup(backupURL, BackupFilePath); err != nil {
+		return err
+	}
+
+	outputFile := c.String("output-file")
+	backingFileOrDir := c.String("backing-file")
+	if backingFileOrDir == "" {
+		if err := ConvertImage(BackupFilePath, outputFile, imageFormat); err != nil {
+			return err
+		}
+	} else {
+		backingFilepath, err := util.ResolveBackingFilepath(backingFileOrDir)
+		if err != nil {
+			return err
+		}
+		if err := CheckBackingFileFormat(backingFilepath); err != nil {
+			return err
+		}
+		if err := CopyFile(backingFilepath, BackingFileCopy); err != nil {
+			return err
+		}
+		if err := ConvertImage(BackupFilePath, BackupFileConverted, DefaultImageFormat); err != nil {
+			return err
+		}
+		if err := MergeSnapshots(BackupFileConverted, BackingFileCopy); err != nil {
+			return err
+		}
+		if err := ConvertImage(BackingFileCopy, outputFile, imageFormat); err != nil {
+			return err
+		}
+	}
+	defer CleanupTempFiles(outputFile, BackupFilePath, BackupFileConverted, BackingFileCopy)
+
+	logrus.Infof("Done running restore to file command. Produced image: %s", outputFile)
+
+	return nil
+}
+
+func CheckBackingFileFormat(backingFilePath string) error {
+	var stdout bytes.Buffer
+	cmd := exec.Command(QEMUImageBinary, "info", backingFilePath)
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	outStr := string(stdout.Bytes())
+	if !strings.Contains(outStr, "file format: qcow2") {
+		return fmt.Errorf("The format of backing file is not qcow2. backing file info=%s", outStr)
+	}
+	return nil
+}
+
+func CopyFile(backingFilepath, outputFile string) error {
+	if err := exec.Command("cp", "-rf", backingFilepath, outputFile).Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CleanupTempFiles(outputfile string, files ...string) error {
+	for _, file := range files {
+		if file == outputfile {
+			continue
+		}
+		if err := exec.Command("rm", "-rf", file).Run(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ConvertImage(srcFilepath, dstFilepath, format string) error {
+	defer logrus.Debugf("ConvertImage (%s) %s -> %s", format, srcFilepath, dstFilepath)
+
+	return exec.Command(QEMUImageBinary, "convert", "-O", format, srcFilepath, dstFilepath).Run()
```

Comment:
Log CombinedOutput for debug purpose.

Also if the error happens, indicate it happens during which stage, e.g. ConvertImage.

Same below.